dotnet3.5/S3/CopyObjectTest/CopyObjectTest.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Moq;
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace CopyObjectTest
+{
+    public class CopyObjectTest
+    {
+        private string _SourceBucket;
+        private string _DestinationBucket;
+        private string _SourceObjectKey;
+        private string _DestinationObjectKey;
+
+        private IAmazonS3 CreateMockS3Client()
+        {
+            var mockS3Client = new Mock<IAmazonS3>();
+            mockS3Client.Setup(client => client.CopyObjectAsync(
+                It.IsAny<CopyObjectRequest>(),
+                It.IsAny<CancellationToken>()
+            )).Callback<CopyObjectRequest, CancellationToken>((request, token) =>
+            {
+                if (!String.IsNullOrEmpty(request.SourceBucket))
+                {
+                    Assert.Equal(request.SourceBucket, _SourceBucket);
+                }
+            }).Returns((CopyObjectRequest r, CancellationToken token) =>
+                {
+                    return Task.FromResult(new CopyObjectResponse()
+                    {
+                        HttpStatusCode = System.Net.HttpStatusCode.OK
+                    });
+                });
+
+            return mockS3Client.Object;
+        }
+
+        [Fact]
+        public async Task CheckCopyObject()
+        {
+            IAmazonS3 client = CreateMockS3Client();
+
+            // Set field values
+            _SourceBucket = "CopyObjectSampleSource";
```

Comment:
For example or "dummy" Amazon S3 bucket names, we're currently only approved to choose from the following: "doc-example-bucket," "doc-example-bucket1," and "doc-example-bucket2."

Reply:
Changed all code to use these names.